src/primitives/compression.cpp

```diff
@@ -0,0 +1,281 @@
+#include <primitives/compression.h>
+
+CCompressedOutPoint::CCompressedOutPoint(const uint256& txid, const uint32_t& vout) : m_block_height(0), m_block_index(0), m_txid(txid), m_vout(vout) {};
+
+CCompressedOutPoint::CCompressedOutPoint(const uint32_t& block_height, const uint32_t& block_index) : m_block_height(block_height), m_block_index(block_index), m_vout(0) {};
```

Comment:
In 86c21e030c2dda625fcc3641266f8eb0fec27173:

It feels weird to pass `uint32_t`s by reference.

Reply:
Fixed, Thanks!